processing/src/main/java/org/apache/carbondata/processing/csvreaderstep/UnivocityCsvParser.java

```diff
@@ -41,6 +41,10 @@
 public class UnivocityCsvParser {
 
   /**
+   * Max number of columns that will be parsed for a row by univocity parsing
+   */
+  private static final int MAX_NUMBER_OF_COLUMNS_FOR_PARSING = 2000;
```

Comment:
I think i will fail if csv file has more number of column than 2000 and schema you have selected less columns. Better expose one property so user can also configure max number of columns in csv file.